clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -58,6 +58,7 @@ public class IsolatedIngestionBackend extends DefaultIngestionBackend
   private final VeniceConfigLoader configLoader;
   private final ExecutorService completionReportHandlingExecutor = Executors.newFixedThreadPool(10);
   private Process isolatedIngestionServiceProcess;
+  private boolean isShuttingDown = false;
```

Comment:
Does it have concurrent accesses? Feels like we might want to use an AtomicBoolean if it happens to.

Reply:
No, the only call is from HelixParticipantService for once during stop(). Each server has only 1 instance of this service class. If we have concurrent access in the future(which I don't think would be true for now) we can evolve to use Atomic I think.

Reply:
I see, but isn't `executeCommandWithRetry` which reads it this value can run from a thread pool, or maybe I misunderstood the way how II works

Reply:
My understanding is: During the shutdown, only HelixParticipantService will modify its value. All Helix threads will read it if it hits this condition (conflict metadata for two processes). Your concern of race condition should not be the case as even if it is missing the update for this round of check it will be caught in the next round of infinite loop and it will break the loop.

Reply:
Thanks for the discussion and updates, Jialin!